nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/Instrumenter.java

```diff
@@ -92,40 +107,94 @@ private Instrumenter() {
-            throws IOException, NoSuchAlgorithmException, FileNotFoundException {
-        ZipFile rt = null;
-        try {
-            rt = new ZipFile(rtJar);
-        } catch (IOException exc) {
-            Logger.getGlobal().log(Level.SEVERE, "Are you sure you provided a valid path to your rt.jar?");
-            throw exc;
+    private InputStream getClassInputStream(String className)
+            throws IOException {
+        InputStream clInputStream;
+        if (rtZipFile != null) {
+            try {
+                ZipEntry entry = rtZipFile.getEntry(className);
+                if (entry == null) {
+                    Logger.getGlobal().log(Level.WARNING, "Could not find " + className + " in rt.jar");
+                    Logger.getGlobal().log(Level.WARNING, "Are you running java 8?");
+                    Logger.getGlobal().log(Level.WARNING, "Continuing without instrumenting: " + className);
+                    return null;
+                }
+                clInputStream = rtZipFile.getInputStream(entry);
+                return clInputStream;
+            } catch (IOException exc) {
+                Logger.getGlobal().log(Level.WARNING, "Cannot find " + className + " are you sure this is a valid rt.jar?");
+                Logger.getGlobal().log(Level.WARNING, "Continuing without instrumenting: " + className);
+            }
         }
-        final Set<String> classesToCopy = this.filterCached(rt, outJar);
 
-        // If no class needs to be reinsturmented
+        if (rtFileSystem != null) {
+            try {
+                Path curClsPath = rtFileSystem.getPath(rootPath, className);
+                byte[] clsBytes = Files.readAllBytes(curClsPath);
+                return new ByteArrayInputStream(clsBytes);
+            } catch (IOException exc) {
+                Logger.getGlobal().log(Level.WARNING, "Could not find " + className + " in jrt file system");
+                Logger.getGlobal().log(Level.WARNING, "Continuing without instrumenting: " + className);
+            }
+        }
+        return null;
+    }
+
+    static boolean hasClassEntry(String className) {
+        if (rtZipFile != null) {
+            return rtZipFile.getEntry(className) != null;
+        } else {
+            Path curClsPath = rtFileSystem.getPath(rootPath, className);
+            return Files.exists(curClsPath);
+        }
+    }
+
+    private void initAndProcess(String rtPath, String outJar)
+            throws IOException, NoSuchAlgorithmException {
+        if (Utils.checkJDKBefore8()) {
+            ZipFile rt = null;
+            try {
+                rt = new ZipFile(rtPath);
+            } catch (IOException exc) {
+                Logger.getGlobal().log(Level.SEVERE, "Are you sure you provided a valid path to your rt.jar?");
+                throw exc;
+            }
+            rtZipFile = rt;
+        } else {
+            rtFileSystem = FileSystems.getFileSystem(URI.create("jrt:/"));
```

Comment:
I assume this is the new part to consider for Java 9? Maybe add a comment to remind us about that.